app/models/cart_item.rb

```diff
@@ -1,8 +1,34 @@
 class CartItem < ApplicationRecord
-  belongs_to :product, optional: true
+  belongs_to :product
   belongs_to :cart, optional: true
 
+  before_save :set_save_price_to_cart_item
+
+  CARTITEM_ATTRS = %i(quantity product_id).freeze
+
   validates :quantity, presence: true,
     numericality:
     {only_integer: true, greater_than: Settings.cart_item.validation.greater}
+
+  def save_price_to_cart_item
+    if persisted?
+      self[:save_price_to_cart_item]
+    else
+      product.price
+    end
+  end
+
+  def get_total_price
+    save_price_to_cart_item * quantity
+  end
+
+  private
+
+  def set_save_price_to_cart_item
+    self[:save_price_to_cart_item] = save_price_to_cart_item
+  end
+
+  def set_total
+    self[:get_total_price] = get_total_price * quantity
```

Comment:
```suggestion
    self.get_total_price = get_total_price * quantity
```

get_total_price là thuộc tính hay là gì thế ?

Reply:
là thuộc tính ạ